### ISSUE TITLE
Gcc47 import fail

### ISSUE THREAD

**Issue by ebobak** [48kRAM/munkiports#1]
Josh,

Not sure if this a packages issue or the import script but trying to import all of my macports it fails and does not proceed after gcc47.

Importing port gcc47
gcc47 (v 4.7.4)
Found pkg for gcc47
Making disk image containing gcc47-1_4.7.4_5.pkg...
hdiutil: create failed - error -5341
Disk image creation failed.
Could not convert /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_release_tarballs_ports_lang_gcc47/gcc47/work/gcc47-1_4.7.4_5.pkg to a disk image.
munkiimport failed!
snootles:Import ebobak$ sudo ./import-macports -r /Users/Shared/munki_repo -p gcc47

I usually have to increase the size to 1gb using flag -size 1gb for hdiutil in Jenkins scrips when we run across this error.

**Comment by 48kRAM** [comment]
Hrm. Could be an internal MacPorts error. What happens when you run `sudo port pkg gcc47` ?

**Comment by 48kRAM** [comment]
Ah - I see why I didn't run across this in my environment: I'm building flat packages.
Set this option in your /opt/local/etc/macports/macports.conf file

```
package.flat            yes
```

and munkiimport won't have to wrap your pkg files in a DMG anymore.

**Comment by 48kRAM** [comment]
To clarify: This is not a munkiports or a MacPorts issue.

This issue appears to be related to https://github.com/munki/munki/issues/273 where hdiutil improperly computes the space needed to build a DMG. Apparently this is something with the hdiutil itself, possibly on Mavericks.
What OS are you running these scripts on?

**Comment by ebobak** [comment]
`sudo port pkg gcc47` works.  Trying with `sudo ./import-macports` still tries to make dmg's out of them.  I'll just make a script to grab gcc and its depen to a flat package.  I'm running on 10.10.5.
I knew I saw that hdiutil error somewhere in the goolge groups and forgot its location.

**Comment by 48kRAM** [comment]
Can you make the change to macports.conf to build flat packages instead?
I've pushed some changes that will detect when that configuration change
needs to be made.

**Comment by ebobak** [comment]
Yes. Updated the config to use package.flat and package_flat bit same thing. I'll have to dive deeper on Wed. 

## 

Ed Bobak

**Comment by 48kRAM** [comment]
We just did an import of gcc49 and this issue isn't showing - at least on OSX 10.10. Since I'd completely forgotten about this issue and it's quite old, I'm inclined to just close it.
